plugins/modules/net_tools/ldap/ldap_search.py

```diff
@@ -0,0 +1,213 @@
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r"""
+---
+module: ldap_search
+short_description: Search for entries in a LDAP server.
+description:
+  - Return the results of an LDAP search. Use Ansible's 'register' statement.
+notes:
+  - Only supports Python3
+  - The default authentication settings will attempt to use a SASL EXTERNAL
+    bind over a UNIX domain socket. This works well with the default Ubuntu
+    install for example, which includes a cn=peercred,cn=external,cn=auth ACL
+    rule allowing root to modify the server configuration. If you need to use
+    a simple bind to access your server, pass the credentials in I(bind_dn)
+    and I(bind_pw).
+version_added: "2.9"
+author:
+  - Peter Sagerson
+  - Sebastian Pfahl
+requirements:
+  - python-ldap
+options:
+  dn:
+    required: true
+    type: str
+    description:
+      - The LDAP DN to search in.
+  scope:
+    required: false
+    choices: [ base, onelevel, subordinate, children ]
+    default: base
+    type: str
+    description:
+      - The LDAP scope to use.
+  filter:
+    required: false
+    default: '(objectClass=*)'
+    type: str
+    description:
+      - Used for filtering the LDAP search result.
+  attrs:
+    required: false
+    default: none
+    type: raw
+    description:
+      - A list of attributes for limiting the result. Use an
+        actual list or a comma-separated string.
+  schema:
+    required: false
+    default: false
+    type: bool
+    description:
+      - Set to True to return the full attribute schema of entries, not
+        their attribute values. Overrides C(attrs) when provided.
+extends_documentation_fragment:
+  - ldap.documentation
+"""
+
+EXAMPLES = r"""
+# Return all entries within the 'groups' organizational unit.
+- ldap_search:
+    dn: "ou=groups,dc=example,dc=com"
+  register: ldap_groups
+
+# Return GIDs for all groups
+- ldap_search:
+    dn: "ou=groups,dc=example,dc=com"
+    scope: "onelevel"
+    attrs: "gidNumber"
+  register: ldap_group_gids
+"""
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib, BOOLEANS
+from ansible.module_utils._text import to_native, to_bytes
+from ansible_collections.community.general.plugins.module_utils.ldap import LdapGeneric, gen_specs
+
+LDAP_IMP_ERR = None
+try:
+    import ldap
+
+    HAS_LDAP = True
+except ImportError:
+    LDAP_IMP_ERR = traceback.format_exc()
+    HAS_LDAP = False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=gen_specs(
+            dn=dict(type='str', required=True),
+            scope=dict(type='str', default='base', choices=['base', 'onelevel', 'subordinate', 'children']),
+            filter=dict(type='str', default='(objectClass=*)'),
+            attrs=dict(type='raw', default=None),
+            schema=dict(type='bool', default='false', choices=(list(BOOLEANS) + ['True', True, 'False', False])),
```

Comment:
I believe this is the right format, you'll want to check. 
```suggestion
            schema=dict(type='bool', default=False, type='bool' ,
```